### PR TITLE
Add support for WFS queries

### PR DESCRIPTION
**Pull request by tsauerwein**
When doing a query with a coordinate, WFS GetFeature will be used if possible, otherwise WMS GetFeatureInfo. Queries with a bbox are made with GetFeature requests.

Still todo:
* [x] Add support for WMTS layers (changes in c2cgeoportal required)
* [x] Make sure previous requests are canceled when a new query is made

Closes https://github.com/camptocamp/c2cgeoportal/issues/1771

### CODE REVIEW

File: contribs/gmf/apps/desktop/js/controller.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ goog.require('gmf.AbstractDesktopController');
 /** @suppress {extraRequire} */
 goog.require('gmf.DatePickerDirective');
 /** @suppress {extraRequire} */
+goog.require('ngeo.bboxQueryDirective');
+/** @suppress {extraRequire} */
```
Comment (sbrunner) [inline contribs/gmf/apps/desktop/js/controller.js]: Should be in the `abstractdesktop.js` file :-)

**Comment by sbrunner** [comment]
Can you provide an URL for tests ?

**Comment by tsauerwein** [comment]
> Can you provide an URL for tests ?

Here is a demo: https://tsauerwein.github.io/ngeo/query-wfs/examples/contribs/gmf/apps/desktop

Currently I am waiting on a fix to get merged in ol3 related  to issues with the compiled version, but the PR is ready to be reviewed.

**Comment by sbrunner** [comment]
I don't succeed to get a query on background layer and on WMTS overlay ...

**Comment by tsauerwein** [comment]
> I don't succeed to get a query on background layer and on WMTS overlay ...

The [themes service](https://geomapfish-demo.camptocamp.net/2.1/wsgi/themes?version=2&background=background) still returns `ogcServer` instead of `serverOGC`. If you could update that part in c2cgeoportal I could test properly.

File: contribs/gmf/apps/desktop/index.html [hunk 2]
```diff
@@ -196,7 +196,7 @@
         module.constant('gmfRasterUrl', 'https://geomapfish-demo.camptocamp.net/2.1/wsgi/raster');
         module.constant('gmfPrintUrl', 'https://geomapfish-demo.camptocamp.net/2.1/wsgi/printproxy');
         module.constant('gmfWmsUrl', 'https://geomapfish-demo.camptocamp.net/2.1/wsgi/mapserv_proxy');
-        module.constant('gmfTreeUrl', 'https://geomapfish-demo.camptocamp.net/2.1/wsgi/themes?version=2&background=background');
+        module.constant('gmfTreeUrl', 'http://localhost:3000/themes.js');
```
Comment (sbrunner) [inline contribs/gmf/apps/desktop/index.html]: Should be removed :-)

Comment (tsauerwein) [inline contribs/gmf/apps/desktop/index.html]: Thanks

**Comment by sbrunner** [comment]
Looks good for me :-)

**Comment by sbrunner** [comment]
Looks with @fredj , you can merge it :-)

**Comment by tsauerwein** [comment]
Thanks for the review!

**Pull request merged**